tests/trainer/test_checkpoint.py

```diff
@@ -230,275 +182,48 @@ def test_ignore_params(remove_field_paths: List[List[str]], filter_params: List[
     assert base_dict == new_dict
 
 
-@device('cpu', 'gpu')
-def test_load_weights(
-    device: str,
-    composer_trainer_hparams: TrainerHparams,
-):
-    """strategy:
-    - train two epochs. capture checkpoints after `save_interval` and ep2.
-    - create a new trainer from the `save_interval` checkpoint, but with a new optimizer and scheduler.
-    - assert that the model weights are the original model, even though the optimizer and scheduler are different.
-    """
-    if not isinstance(composer_trainer_hparams.train_dataset, SyntheticHparamsMixin):
-        pytest.skip('Checkpointing tests require synthetic data')
-        return
-    if not isinstance(composer_trainer_hparams.val_dataset, SyntheticHparamsMixin):
-        pytest.skip('Checkpointing tests require synthetic data')
-        return
-    checkpoint_a_folder = 'first'
-    final_checkpoint = 'ep2.pt'
-    composer_trainer_hparams = get_two_epoch_composer_hparams(composer_trainer_hparams, checkpoint_a_folder)
-
-    second_trainer_hparams = copy.deepcopy(composer_trainer_hparams)
-    _test_checkpoint_trainer(composer_trainer_hparams)
-
-    # Reduce the filepath to get the location on the rank zero process
-    checkpoint_a_file_path = [os.path.join(os.path.abspath(checkpoint_a_folder), final_checkpoint)]
-    dist.broadcast_object_list(checkpoint_a_file_path)
-
-    # load only model weights
-    second_trainer_hparams.load_path = checkpoint_a_file_path[0]
-    second_trainer_hparams.load_weights_only = True
-    second_trainer_hparams.load_strict_model_weights = True
-    # setup a new optimizer
-    second_trainer_hparams.optimizers = AdamWHparams()
-
-    # setup a new LR scheduler
-    assert isinstance(second_trainer_hparams.max_duration, str)
-    second_trainer_hparams.schedulers = [CosineAnnealingScheduler(t_max=second_trainer_hparams.max_duration)]
```

Comment:
I removed the check for the different scheduler type, as I wasn't sure what its purpose is in this test.. does keeping the same scheduler somehow change how the weights are loaded?

Reply:
I believe it's to test `load_weights` doesn't override other params, eg scheduler? I'm not super sure. it might be nice to include in the test though to verify `load_weights` doesn't touch other parts

Reply:
Good point, I added a new test to check that the statefulcallback was not touched.